### PR TITLE
SOLR-16664: TestCoordinatorRole fails docs is null

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16664

I started to look at `TestCoordinatorRole` and besides just adding the assertion to make sure docs is not null there are other confusing things in this test.

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java [hunk 1]
```diff
@@ -389,6 +390,7 @@ private String getHostCoreName(
     SolrQuery q = new SolrQuery("*:*");
     q.add("fl", "id,desc_s,_core_:[core]").add(OMIT_HEADER, TRUE);
     p.accept(q);
+    // TODO why is this building a string instead of using the proper SolrClient???
```
Comment (noblepaul) [inline solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java:393]: SolrClient will not send the request to COORDINATOR , so we can't  test the behaviour

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java [hunk 2]
```diff
@@ -296,6 +294,8 @@ public void testNRTRestart() throws Exception {
         assertEquals(nrtCore, hostCore);
         // allow any exceptions to propagate
         jettyManipulationFuture.get();
+
+        // TODO what is expected here? why are we just bailing? Can't the following code just be removed...
```
Comment (noblepaul) [inline solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java:298]: this got accidentally added while debugging. this line must go

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java [hunk 3]
```diff
@@ -338,6 +338,7 @@ public void testNRTRestart() throws Exception {
         start = new Date().getTime();
         try {
           do {
+            // TODO - What is expected here? equals not doing anything? hostCore=?
```
Comment (noblepaul) [inline solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java:341]: it should be this
```
      if( pullCore.equals(
                    getHostCoreName(
                        COLL,
                        qaJettyBase,
                        client,
                        p -> {
                          p.set(CommonParams.Q, "id:345");
                          p.add("shards.preference", "replica.type:NRT");
                        }))){
              done.set(true);
            }
```

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java (hunk 1 above)
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: I don't understand what you mean. I pushed up ff1f547eada49b494f5943ff2bb0e95ef3f4a671 which does exactly what I said using SolrClient.

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java (hunk 3 above)
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: Fixed in e409355e572f08491b199841f773166e0b4c7b3c I think.

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java (hunk 2 above)
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: Removed in 869ee24dcfb1fbed253291d2f49945d186eaccda

File: solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java (hunk 1 above)
Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: if you use `SolrClient` how do you ensure that the request is actually going to coordinator node?

Comment (risdenk) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: Unless I misunderstand how this works `try (SolrClient solrClient = new Http2SolrClient.Builder(qaNode).build()) {` - this builds a SolrClient pointing to the qanode.

Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: >this builds a SolrClient pointing to the qanode.

that is not what it does. `Http2SolrClient.Builder(qaNode).build()` builds a client with the `qaNode` as the "seed node". Thereafter, it queries the `qaNode` to get the collection topology and finds out where the request should be sent. Basically, it behaves the similar to the Zk based provider but using the REST APIs

Comment (HoustonPutman) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: Are you sure you don't mean `CloudHttp2SolrClient`? I don't think that the `Http2SolrClient` should know anything about the cloud topology.

Comment (noblepaul) [reply solr/core/src/test/org/apache/solr/search/TestCoordinatorRole.java]: Oh, my bad. You are right . I confused it with `CloudHttp2SolrClient`

**Review by noblepaul — APPROVED** [review]

**Pull request merged**